libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/Gov.hs

```diff
@@ -4,35 +4,44 @@
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE QuasiQuotes #-}
 {-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE ViewPatterns #-}
 
 -- | Specs necessary to generate, environment, state, and signal
 -- for the GOV rule
 module Test.Cardano.Ledger.Constrained.Conway.Gov where
 
-import Cardano.Ledger.Shelley.HardForks qualified as HardForks
-import Data.Foldable
-
-import Data.Coerce
-
 import Cardano.Ledger.BaseTypes
 import Cardano.Ledger.CertState
+import Cardano.Ledger.Conway (Conway, ConwayEra)
+import Cardano.Ledger.Conway.Core
 import Cardano.Ledger.Conway.Governance
-import Cardano.Ledger.Conway.PParams
 import Cardano.Ledger.Conway.Rules
+import Cardano.Ledger.Crypto (StandardCrypto)
+import Cardano.Ledger.Shelley.HardForks qualified as HardForks
+import Constrained
+import Constrained.Base (Pred (..))
+import Data.Coerce
+import Data.Foldable
 import Data.Map qualified as Map
 import Data.Set qualified as Set
 import Lens.Micro
-
-import Constrained
-
-import Cardano.Ledger.Conway (ConwayEra)
-import Cardano.Ledger.Conway.Core
-import Cardano.Ledger.Crypto (StandardCrypto)
 import Lens.Micro qualified as L
 import Test.Cardano.Ledger.Constrained.Conway.Instances
 import Test.Cardano.Ledger.Constrained.Conway.PParams
+import Test.Cardano.Ledger.Generic.PrettyCore (PrettyA (..))
+import Test.QuickCheck hiding (forAll)
+
+main :: IO ()
+main = do
+  env <- generate $ genFromSpec @ConwayFn govEnvSpec
+  let spec = govProposalsSpec @ConwayFn env
+  p <- generate (vectorOf 20 (genFromSpec spec))
+  case p of
+    (y : _) -> putStrLn $ (show . prettyA) y
+    _ -> pure ()
+  putStrLn $ show (map (\x -> conformsToSpec x spec) p)
```

Comment:
There should ever be one `main` and it is not in the library
```suggestion
```

Reply:
done